### PR TITLE
Add delete privilege to kibana_system for APM

### PR DESCRIPTION
**Pull request by axw**
Extend the `kibana_system` privileges to allow deletion of all APM indices.

This is a followup to https://github.com/elastic/elasticsearch/pull/81811. Delete phases were added to all APM data stream ILM policies in 8.0.

Closes https://github.com/elastic/kibana/issues/128014 https://github.com/elastic/apm-server/issues/7568

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java [hunk 1]
```diff
@@ -770,7 +770,17 @@ public static RoleDescriptor kibanaSystemRoleDescriptor(String name) {
...
+                    .indices(
+                        ".logs-endpoint.diagnostic.collection-*",
+                        "logs-apm.app-*",
+                        "logs-apm.error-*",
+                        "metrics-apm.app.*",
+                        "metrics-apm.internal-*",
+                        "metrics-apm.profiling-*",
+                        "traces-apm-*",
+                        "traces-apm.rum-*",
+                        "traces-apm.sampled-*"
```
Comment (simitt) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:782]: Do we need to be that specific? If we went ahead with `logs-apm.*`,  `logs-apm-*`, `traces-apm.*`, `traces-apm-*` ,  `metrics-apm.*` and `metrics-apm-*` we'd lower the risk of missing data streams in the future and be more aligned with [current definitions](https://github.com/elastic/elasticsearch/blob/ce5c63508a4b0da52bd4a6e8dd289bb2bf656431/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java#L257-L274).

Comment (axw) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:782]: Good call. I'll update to that.

**Comment by axw** [comment]
@elasticmachine run elasticsearch-ci/part-1

**Review by simitt — APPROVED** [review]

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java [hunk 2]
```diff
@@ -770,7 +770,15 @@ public static RoleDescriptor kibanaSystemRoleDescriptor(String name) {
...
                 RoleDescriptor.IndicesPrivileges.builder()
-                    .indices(".logs-endpoint.diagnostic.collection-*", "traces-apm.sampled-*")
+                    .indices(
+                        ".logs-endpoint.diagnostic.collection-*",
+                        "logs-apm-*",
+                        "logs-apm.*",
+                        "metrics-apm-*",
+                        "metrics-apm.*",
+                        "traces-apm-*",
+                        "traces-apm.*"
```
Comment (joshdover) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:780]:
```suggestion
                        ".logs-endpoint.diagnostic.collection-*",
                        "logs-apm-*",
                        "logs-apm.*-*",
                        "metrics-apm-*",
                        "metrics-apm.*-*",
                        "traces-apm-*",
                        "traces-apm.*-*"
```

**Review by joshdover — APPROVED** [review]
One nit on the index pattern, other LGTM

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java (hunk 2 above)
Comment (joshdover) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:780]: Probably will need to adjust the tests for this too

**Comment by ywangd** [comment]
Ping @elastic/kibana-security for awareness

**Review by jportner — APPROVED** [review]
> Ping @elastic/kibana-security for awareness

Thanks! I reviewed the issues and the changes make sense to me.
Approving this with a LGTM for posterity :+1:

**Review by ywangd — APPROVED** [review]
LGTM

**Comment by axw** [comment]
Thanks all!

**Comment by axw** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.0|[](https://github.com/elastic/elasticsearch/pull/85200)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**